File: src/main/java/org/ffpy/plugin/coding/constant/TemplateName.java
package org.ffpy.plugin.coding.constant;

/**
 * 模板名称
 */
public enum TemplateName {
    ENUM_CODE("EnumCode.vm"),
    ;

    public static final String PATH_TEMPLATE = "/template/";

    public static String getPath(String name) {
        return PATH_TEMPLATE + name;
    }

    private final String name;

    TemplateName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public String getPath() {
        return PATH_TEMPLATE + name;
    }
}

File: src/main/kotlin/org/ffpy/plugin/coding/constant/TemplateName.kt
package org.ffpy.plugin.coding.constant

/**
 * 模板名称
 */
enum class TemplateName(val fileName: String) {
    ENUM_CODE("EnumCode.vm"),
    XML_TO_BEAN("XmlToBean.vm");

    val path: String
        get() = PATH_TEMPLATE + fileName

    companion object {
        const val PATH_TEMPLATE = "/template/"
    }
}